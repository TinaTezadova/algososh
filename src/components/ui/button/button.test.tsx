import { Button } from './button';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Тестирвание компонента Button', () => {
    it('Рендер кнопки с текстом', () => {
        render(<Button text="Текст" />);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot();
    });

    it('Рендер кнопки без текста', () => {
        render(<Button />);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot();
    });

    it('Рендер заблокированной кнопки', () => {
        render(<Button disabled />);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot();
    });

    it('Рендер кнопки с лоадером', () => {
        render(<Button isLoader />);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot();
    });

    it('Проверка клика на кнопку', () => {
        const click = jest.fn();
        render(<Button onClick={click} />);
        const button = screen.getByRole('button');

        expect(click).toBeCalledTimes(0);
        userEvent.click(button);

        expect(click).toHaveBeenCalledTimes(1);
    });

});