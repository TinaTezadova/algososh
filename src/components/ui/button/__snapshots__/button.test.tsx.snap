// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирвание компонента Button Рендер заблокированной кнопки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Тестирвание компонента Button Рендер кнопки без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Тестирвание компонента Button Рендер кнопки с лоадером 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирвание компонента Button Рендер кнопки с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Текст
  </p>
</button>
`;
